src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -241,4 +255,12 @@ public void setDeadlineTaskFilter(Predicate<? super DeadlineTask> predicate) {
         filteredDeadlineTasks.setPredicate(predicate);
     }
 
+    @Override
+    public void refilterDeadlineTaskFilteredList() {
+        Predicate<? super DeadlineTask> predicate = filteredDeadlineTasks.getPredicate();
+        assert predicate != null;
+        setDeadlineTaskFilter(null);
+        setDeadlineTaskFilter(predicate);
```

Comment:
> But if we just create a new task every time we want to finish it, this function will be just the same as edit.

What's wrong with that?

I mean, if you think about it, *technically* it is the same. Both are editing the fields of a task.